### PR TITLE
[STK-220] add initial code structure

### PR DESCRIPTION
**Pull request by kahootali**

### CODE REVIEW

File: internal/pkg/controller/controller.go [hunk 1]
```diff
@@ -1,25 +1,60 @@
...
-
+	for {
+		nameSpaceList, err := c.clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
+		if err != nil {
+			log.Printf("Error getting namespaces: %v", err)
+			return
+		}
+		for _, namespace := range nameSpaceList.Items {
+			annotations := namespace.Annotations
+			if value, ok := annotations[jamadaarDisableAnnotation]; ok {
```
Comment (rasheedamir) [inline internal/pkg/controller/controller.go]: @kahootali plz ensure it should cover both cases:

1. it doesn't exist
2. its value is false

File: internal/pkg/controller/controller.go [hunk 2]
```diff
@@ -1,25 +1,60 @@
...
+//Run function for controller which handles the logic
 func (c *Controller) Run() {
-
+	for {
+		nameSpaceList, err := c.clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
+		if err != nil {
+			log.Printf("Error getting namespaces: %v", err)
+			return
+		}
+		for _, namespace := range nameSpaceList.Items {
```
Comment (rasheedamir) [inline internal/pkg/controller/controller.go]: @kahootali plz create a separate unit testable method which just handles this particular jamaadar task of the the cleanup of namespaces; and then write unit test for it

I mean pass it a list of namespaces so, that you can test it's core logic without running it inside k8s

this controller will have lots of other tasks of cleanup so, proper design of the solution is very important; maybe have a strategy interface and then each cleanup task should implement it?

File: internal/pkg/actions/slack/slack.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package slack
```
Comment (rasheedamir) [inline internal/pkg/actions/slack/slack.go:1]: @kahootali  isn't this a copy paste of the one in Chowkidar? So, should we move into a common package and use in both?

**Comment by rasheedamir** [comment]
@kahootali few more suggestions:

1. give if someone doesn't use slack but still want to use Jamadaar then it must work; and in that case default always enabled action that is "logging to console" should work; create "log to console" as an action.
2. since this delete namespaces is first task of the "Jamadaar" so, it should be implemented in a way so, that we can easily add in pluggable/extendable way more cleanup tasks; I mean anyone can fork add new cleanup task as plugin and create PR and thats it. So, the controller picks up all cleanup tasks from a directory and just performs them; we can discuss it over call as well if its confusing.

**Comment by stakater-user** [comment]
@kahootali Yikes! You better fix it before anyone else finds out! [Build 1](https://jenkins.tools.stackator.com/job/Stakater/job/Jamadaar/job/PR-1/1/) has Failed!

**Comment by stakater-user** [comment]
@kahootali Image is available for testing. `docker pull stakater/jamadaar:PR-1-2`

File: configs/config.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+polltimeinterval: 20s    # Values: "ms", "s", "m", "h".
+age: 7d                  # Values: "d" or "w" or "m" or "y".
```
Comment (hazim1093) [inline configs/config.yaml]: In future we should specify criteria as well, like on "namespaces" or pods etc

File: glide.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  version: kubernetes-1.8.0
+- package: k8s.io/client-go
+  version: 5.0.0
+- package: gopkg.in/yaml.v2
+- package: github.com/spf13/cobra
+- package: github.com/spf13/pflag
+  version: 583c0c0531f06d5278b7d917446061adc344b5cd
+# TODO: Change this back to nlopes when that issue is fixed 
+# URL: https://github.com/nlopes/slack/issues/294
+- package: github.com/asiyani/slack
```
Comment (hazim1093) [inline glide.yaml:17]: add versions to these packages

**Review by hazim1093 — APPROVED** [review]

**Review by hazim1093 — CHANGES_REQUESTED** [review]
Add versions in glide.yaml

File: configs/config.yaml (hunk 4 above)
Comment (rasheedamir) [reply configs/config.yaml]: @kahootali we should fix this right now; its important that we have well defined criteria

**Review by hazim1093 — APPROVED** [review]

**Pull request merged**